packages/component/src/hooks/useEmojiFromStyles.js

```diff
@@ -0,0 +1,51 @@
+/*eslint no-useless-escape: "off"*/
+/*eslint require-unicode-regexp: "off" */
+import useStyleOptions from './useStyleOptions';
+
+function escapeRegexp(emoticon) {
+  return emoticon.replace(/[\\^$*+?.()|[\]{}]/g, '\\$&');
+}
+
+export default function useEmojiFromStyles() {
+  const [{ emojiAutocorrect: autocorrect, emojiList: customEmojiList }] = useStyleOptions();
+
+  const emojiAutocorrect = autocorrect || false;
+
+  const emojiUnicodeMap = customEmojiList || {
+    ':)': '😊',
+    ':-)': '😊',
+    '(:': '😊',
+    '(-:': '😊',
+    ':-|': '😐',
+    ':|': '😐',
+    ':-D': '😀',
+    ':D': '😀',
+    ':-p': '😛',
+    ':p': '😛',
+    ':-P': '😛',
+    ':P': '😛',
+    ':-o': '😲',
+    ':o': '😲',
+    ':O': '😲',
+    ':-O': '😲',
+    ':-0': '😲',
+    ':0': '😲',
+    ';-)': '😉',
+    ';)': '😉',
+    '<3': '❤️',
+    '</3': '💔',
+    '<\\3': '💔'
```

Comment:
Please add two tests.

```js
{
  ':o': '😲',
  ':o)': '🤡'
}
```

And

```js
{
  ':o)': '🤡',
  ':o': '😲'
}
```

FYI, key order is random and not guaranteed. It is up to the JS engine to decide which one come first, and the dev has no control over it.

If the dev want to have control, we need to use array.